File: Practica1/Ejercicio2/FactoriaCarretera.py
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from CarreraCarretera import CarreraCarretera
from BicicletaCarretera import BicicletaCarretera

class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crear_carrera(self, nombre, bicicletas):
        carrera_de_carretera = CarreraCarretera(nombre, bicicletas)
        return carrera_de_carretera
    
    def crear_bicicleta(self, id):
        bicicleta_de_carretera = BicicletaCarretera(id)
        return bicicleta_de_carretera

File: Practica1/Ejercicio1/src/FactoriaCarretera.java
import java.util.ArrayList;

public class FactoriaCarretera extends Thread implements FactoriaCarreraYBicicleta {
    
    public FactoriaCarretera() {

    }

    public CarreraCarretera crearCarrera(String nombre, ArrayList<Bicicleta> bicicletas) {
        CarreraCarretera objCarreraCarretera = new CarreraCarretera(nombre, bicicletas);

        return objCarreraCarretera;
    }

    public BicicletaCarretera crearBicicleta(int id) {
        BicicletaCarretera objBicicletaCarretera = new BicicletaCarretera(id);

        return objBicicletaCarretera;
    }
}